docs/rules/prefer-class-fields.md

```diff
@@ -0,0 +1,41 @@
+# Prefer class field declarations over `this` assignments in constructor
+
+💼 This rule is enabled in the ✅ `recommended` [config](https://github.com/sindresorhus/eslint-plugin-unicorn#preset-configs-eslintconfigjs).
+
+🔧 This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).
+
+<!-- end auto-generated rule header -->
+<!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
+
+This rule enforces the use of class field declarations for static values, instead of assigning them in constructors using `this`.
+
+> To avoid leaving empty constructors after autofixing, use the [`no-useless-constructor` rule](https://eslint.org/docs/latest/rules/no-useless-constructor).
+
+## Fail
+
+```js
+class Foo {
+	constructor() {
+		this.foo = 'foo';
+	}
+}
+
+class MyError extends Error {
+	constructor(message: string) {
+		super(message);
+		this.name = 'MyError';
+	}
+}
+```
+
+## Pass
+
+```js
+class Foo {
+	foo = 'foo';
+}
+
+class MyError extends Error {
+	name = 'MyError'
+}
+```
```

Comment:
Can you use this style for the examples?

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/4e539b4b084ce06165a0fdbc02d309ce7cf75c82/docs/rules/consistent-existence-index-check.md?plain=1#L16-L36

Reply:
Sure, sorry for that - previously I've looked to [prefer-switch docs](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-switch.md) for inspiration.
Change done in: e839988df76908e9935e782884401bf9d7b3503e